kubernetes: Set default aggregation level to maximum

tgraf opened the pull request:
The BPF event aggregation has been around or a while but was not enabled in the
ConfigMap by default. We have been running this for a long while so it should
be safe to enable by default and have everybody benefit from reduced CPU
consumption while running cilium monitor.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/8115)
<!-- Reviewable:end -->

joestringer reviewed (approved):
LGTM, with the following caveat:
* "maximum" semantically means "apply all optimizations". It could change in future, assuming we introduced additional aggregation logic such as a "high" level. Users would immediately adopt all new behaviour if we set this as the default.
* "medium" has a fixed meaning, "only send notifcation if there is new connection state [eg TCP flags], or if a notification was not sent in the last aggregation interval [default 5s]". The meaning of this should not be redefined, so the behaviour should be retained across all subsequent releases.

Practically they're the same right now, but I introduced both options to allow users to either opt into a known specific implementation now, or opt into the maximum set of optimizations in any given release, which may change over time.

jrfastab reviewed (approved):
:+1:

tgraf committed:
kubernetes: Set default aggregation level to maximum

The BPF event aggregation has been around or a while but was not enabled in the
ConfigMap by default. We have been running this for a long while so it should
be safe to enable by default and have everybody benefit from reduced CPU
consumption while running cilium monitor.

Signed-off-by: Thomas Graf <thomas@cilium.io>
